diff --git a/www/app/controllers/PageController.php b/www/app/controllers/PageController.php
new file mode 100644
index 0000000..6e337f7
--- /dev/null
+++ b/www/app/controllers/PageController.php
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace app\controllers;
+
+/** @property \app\models\Page $model */
+
+class PageController extends AppController
+{
+    public function viewAction()
+    {
+        $lang = \wfm\App::$app->getProperty('language');
+        $page = $this->model->get_page($this->route['slug'], $lang);
+        if (is_null($page) || empty($page)) {
+            $this->error_404();
+            return;
+        }
+
+        $this->setMeta($page['title'], (string) $page['description'], (string) $page['keywords']);
+        $this->set(compact('page'));
+    }
+}
diff --git a/www/app/models/Page.php b/www/app/models/Page.php
new file mode 100644
index 0000000..3703040
--- /dev/null
+++ b/www/app/models/Page.php
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace app\models;
+
+class Page extends AppModel
+{
+    public function get_page(string $slug, array $lang): ?array
+    {
+        $sql = "SELECT p.*, pd.*
+            FROM page p
+            JOIN page_description pd
+                ON p.id = pd.page_id
+            WHERE p.slug = ? 
+                AND pd.language_id = ?";
+
+        $query = \RedBeanPHP\R::getRow($sql, [$slug, $lang['id']]);
+
+        return $query;
+    }
+}
diff --git a/www/app/views/Page/view.php b/www/app/views/Page/view.php
index b3d0e20..86e869d 100644
--- a/www/app/views/Page/view.php
+++ b/www/app/views/Page/view.php
@@ -1,14 +1,14 @@
 <?php
 
 /** 
- * 
+ * @var array $page
  * */
 ?>
 <div class="container">
     <nav aria-label="breadcrumb">
         <ol class="breadcrumb bg-light p-2">
-            <li class="breadcrumb-item"><a href="base_url()"><i class="fas fa-home"></i></a></li>
-            <li class="breadcrumb-item">'title'></li>
+            <li class="breadcrumb-item"><a href="<?= base_url() ?>"><i class="fas fa-home"></i></a></li>
+            <li class="breadcrumb-item"><?= $page['title'] ?></li>
         </ol>
     </nav>
 </div>
@@ -17,11 +17,11 @@
     <div class="row">
 
         <div class="col-lg-12 category-content">
-            <h3 class="section-title">'title'</h3>
+            <h3 class="section-title"><?= $page['title'] ?></h3>
 
-            'content'
+            <?= $page['content'] ?>
 
         </div>
 
     </div>
-</div>
+</div>
\ No newline at end of file
diff --git a/www/app/views/parts/footer.php b/www/app/views/parts/footer.php
index 8b63074..b390b90 100644
--- a/www/app/views/parts/footer.php
+++ b/www/app/views/parts/footer.php
@@ -1,5 +1,7 @@
 <?php
 
+use app\widgets\page\Page;
+
 /** @var wfm\View $this */
 ?>
 <footer>
@@ -8,12 +10,17 @@
             <div class="row">
                 <div class="col-md-3 col-6">
                     <h4><?= ___('tpl_information') ?></h4>
-                    <ul class="list-unstyled">
-                        <li><a href="#">Главная</a></li>
-                        <li><a href="#">О магазине</a></li>
-                        <li><a href="#">Оплата и доставка</a></li>
-                        <li><a href="#">Контакты</a></li>
-                    </ul>
+                    <?php
+                    $prepend = sprintf(
+                        '<li><a href="%s">%s</a></li>',
+                        base_url(),
+                        ___('tpl_home_link')
+                    );
+                    new Page([
+                        'cache' => 0,
+                        'class' => 'list-unstyled',
+                        'prepend' => $prepend,
+                    ]) ?>
                 </div>
 
                 <div class="col-md-3 col-6">
diff --git a/www/app/widgets/page/Page.php b/www/app/widgets/page/Page.php
new file mode 100644
index 0000000..63ecfb6
--- /dev/null
+++ b/www/app/widgets/page/Page.php
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace app\widgets\page;
+
+class Page
+{
+    protected array $language;
+    protected string $container = 'ul';
+    protected string $class = 'page-menu';
+    protected int $cache = 3600;
+    protected string $cachekey = 'ishop_page_menu';
+    protected string $menuPageHtml = '';
+    protected string $prepend = '';
+
+    public function __construct($options = [])
+    {
+        $this->language = \wfm\App::$app->getProperty('language');
+        $this->getOptions($options);
+
+        $cache = \wfm\Cache::getInstance();
+        $cachekey = sprintf("%s_%s", $this->cachekey, $this->language['code']);
+        $html = $cache->get($cachekey);
+        if (!is_string($html) || empty($html)) {
+            $sql = "SELECT p.*, pd.*
+                FROM page p
+                JOIN page_description pd
+                    ON p.id = pd.page_id
+                WHERE pd.language_id = ?";
+
+            $query = \RedBeanPHP\R::getAssoc($sql, [$this->language['id']]);
+            $html = $this->getMenuPageHtml($query);
+
+            if ($this->cache > 0) {
+                $cache->set($cachekey, $html, $this->cache);
+            }
+        }
+        $this->menuPageHtml = $html;
+        $this->output();
+    }
+
+    protected function getMenuPageHtml(array $list): string
+    {
+        $html = '';
+        foreach ($list as $item) {
+            $html .= sprintf(
+                '<li><a href="page/%s">%s</a></li>',
+                $item['slug'],
+                $item['title']
+            ) . PHP_EOL;
+        }
+        return $html;
+    }
+
+    protected function getOptions(array $options): void
+    {
+        foreach ($options as $key => $option) {
+            if (property_exists($this, $key)) {
+                $this->$key = $option;
+            }
+        }
+    }
+
+    protected function output(): void
+    {
+        $html = sprintf('<%s class="%s">', $this->container, $this->class) . PHP_EOL;
+        $html .= sprintf('%s', $this->prepend) . PHP_EOL;
+        $html .= sprintf('%s', $this->menuPageHtml);
+        $html .= sprintf('</$s>', $this->container) . PHP_EOL;
+        echo $html;
+    }
+}
diff --git a/www/config/routes.php b/www/config/routes.php
index 0db97af..f03dcbf 100644
--- a/www/config/routes.php
+++ b/www/config/routes.php
@@ -10,6 +10,7 @@ Router::add('^(?<lang>[a-z]+)?/?product/(?<slug>[a-z0-9-]+)/?$', ['controller' =
 Router::add('^(?<lang>[a-z]+)?/?category/(?<slug>[a-z0-9-]+)/?$', ['controller' => 'Category', 'action' => 'view']);
 Router::add('^(?<lang>[a-z]+)?/?search/?$', ['controller' => 'Search', 'action' => 'index']);
 Router::add('^(?<lang>[a-z]+)?/?wishlist/?$', ['controller' => 'Wishlist', 'action' => 'index']);
+Router::add('^(?<lang>[a-z]+)?/?page/(?<slug>[a-z0-9-]+)/?$', ['controller' => 'Page', 'action' => 'view']);
 Router::add('^(?P<lang>[a-z]+)?/?$', ['controller' => 'Main', 'action' => 'index']);
 Router::add('^(?P<controller>[a-z-]+)/(?P<action>[a-z-]+)/?$');
 Router::add('^(?P<lang>[a-z]+)/(?P<controller>[a-z-]+)/(?P<action>[a-z-]+)/?$');
